### PR TITLE
Christian collect statuscodes

### PR DESCRIPTION
**Pull request by Jowe1999**
This is the statuscodes collection script written in python

### CODE REVIEW

File: igcollect/statuscode2gaphite.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    #search_time = datetime.datetime.utcnow()
+    search_time = datetime.datetime.strptime('09/Mar/2020:10:00:33', '%d/%b/%Y:%H:%M:%S')
+
+    errors = {}
+
+    for i in range(0, 60 * args.time):
+        temp_search_time = (search_time - datetime.timedelta(
+            seconds=i)).strftime('%d/%b/%Y:%H:%M:%S')
+
+        resaults = cmdline('grep {} {} | awk {}| sort | uniq -c'.format(
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: Let's stick to Python.

And even if not, usage of both grep and awk is unnecessary and you're not matching date on date fields but anywhere in line.

File: igcollect/statuscode2gaphite.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from subprocess import run, PIPE
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to collect statuscodes from nginx')
+    parser.add_argument('-l', '--logfile',
+                        help='path to logfile')
+    parser.add_argument('-s', '--software', default='nginx',
+                        help='witch tool is using it (example: nginx)')
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: Spelling!

File: igcollect/statuscode2gaphite.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import argparse
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: Shebang and licence are missing.

File: igcollect/statuscode2gaphite.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                             'has to be the same time as the corresponding '
+                             'cronjob! '
+                             '(default= 1)')
+    parser.add_argument('-p', '--prefix',
+                        default='servers.{}.software.{}'.format(
+                            socket.gethostname().replace('.', '_'),
+                            parser.parse_args().software),
+                        help='the path to the value in Graphite '
+                             '(default= servers.(hostname).software.'
+                             '(software).status_codes)')
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: You're duplicating defaults. It looks like argparse can handle this for you: https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter

File: igcollect/statuscode2gaphite.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    for key in errors.keys():
+        print(template.format('.status_codes', key, errors[key], prob_time))
+        total += errors[key]
+
+    print(template.format('', 'requests', total, prob_time))
+
+
+
+if __name__ == '__main__':
+    main()
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: Missing newline.

File: igcollect/statuscode2gaphite.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    return parser.parse_args()
+
+
+def cmdline(command):
+    process = run(
+        command,
+        stdout=PIPE,
+        stderr=PIPE,
+        universal_newlines=True,
+        shell=True
```
Comment (TuxPowered42) [inline igcollect/statuscode2gaphite.py]: Running things in shell is not really secure, there are ways to inject commands into it.

File: igcollect/statuscode2gaphite.py (hunk 1 above)
Comment (Jowe1999) [reply igcollect/statuscode2gaphite.py]: Ok I look into that

**Pull request closed without merging**